python/ray/rllib/models/action_dist.py

```diff
@@ -102,12 +102,12 @@ def sample(self):
 class MultiCategorical(ActionDistribution):
     """Categorical distribution for discrete action spaces."""
 
+    # noinspection PyMissingConstructor
     def __init__(self, inputs):
-        self.cats = []
-        for input in inputs:
-            self.cats.append(Categorical(input))
+        self.cats = [Categorical(input_) for input_ in inputs]
 
     def logp(self, actions):
+        actions = tf.unstack(actions, axis=1)
```

Comment:
Hmmm, I'm not sure. Think about whether it is more general API that receives a list of tensors or tensor of this shape, regardless of our current usage (would it be more general / better code if the unstack would be executed in the caller).

Reply:
Paging @nikola-j